mirai-core-api/src/commonMain/kotlin/auth/BotAuthorization.kt

```diff
@@ -101,8 +104,115 @@ public interface BotAuthInfo {
     public val id: Long
     public val deviceInfo: DeviceInfo
     public val configuration: BotConfiguration
+
+    /**
+     * 是否是首次登录
+     *
+     * 首次登录指的是首次调用 [Bot.login] 进行登录，直到登录成功的过程。
+     *
+     * 若在首次登录过程中多次进入[认证流程][BotAuthorization.authorize]，则这流程些均被视为首次登录。
+     *
+     * @see BotAuthorization.authorize
+     */
     public val isFirstLogin: Boolean
-    public val reLoginCause: Throwable?
+
+    /**
+     * 导致重新进入认证流程的原因
+     *
+     * 重新进入认证流程一般是第二次或多次进入认证流程，除了 [FastLoginError]。
+     *
+     * 第一次进入认证流程时该值为 `null`
+     *
+     */
+    public val reAuthCause: ReAuthCause?
+}
+
+/**
+ * 导致进行重新与服务器认证的原因
+ *
+ * @see ForceOffline
+ * @see MsfOffline
+ * @see FastLoginError
+ * @see NetworkError
+ * @see Unknown
+ */
+@NotStableForInheritance
+public sealed class ReAuthCause(
+    public open val bot: Bot,
+    public open val message: String?
+) {
+    /**
+     * Bot 被挤下线
+     *
+     * 当 Bot 在其他客户端使用相同协议登录此账号时，会导致 Bot 掉线。
+     *
+     * 若设置了 [BotConfiguration.autoReconnectOnForceOffline] 则 mirai 会尝试重新登录。
```

Comment:
如果有 `ForceOffline` 这个 cause 传给用户那这个属性肯定就是 `true` 吧
```suggestion
     * 会因此原因进行登录就说明 [BotConfiguration.autoReconnectOnForceOffline] 为 `true`。若该属性为 `false` 时则 mirai 不会自动在被挤下线时重新登录。
```

Reply:
我觉得这个描述不太好，因为这里是 `authorize` 过程而不是 `login` 过程，只需要说明 ForceOffline 导致的 authorize 原因即可